### DISCUSSION TITLE
Questions about varying scaling factors in CKKS Bootstrap - understanding why they differ across levels

### DISCUSSION THREAD

**Discussion by Tokubara** [tuneinsight/lattigo#522] [Q&A]
Hello everyone, I have some questions about CKKS Bootstrap. I used to think that the scaling factor $\Delta$ remains constant throughout the CKKS computation process, equal to `LogDefaultScale`. However, after reading the documentation for `bootstrap.ParametersLiteral` regarding `CoeffsToSlotsFactorizationDepthAndLogPlaintextScales`, `SlotsToCoeffsFactorizationDepthAndLogPlaintextScales`, and `EvalModLogPlaintextScale`, I realized my understanding was completely wrong. The scaling factors used in Bootstrap computation are different from the regular scaling factor (`LogDefaultScale`), and moreover, the scaling factors can be different for each level. This confuses me greatly - why can scaling factors change during computation, and why can they even be different for each level? Could you please recommend some references that would help me understand this better?

**Comment by Pro7ech** [comment]
The default scale defines the default value of the field `Scale` when allocating `rlwe.Plaintext`. It is true that it is possible to have the scale be constant (or very close to the default value) across the entire computation (by setting the same bit-size for all primes), although this is tricky in practice when using RNS because the rescaling factors are the primes, which are not divisors of the default scale scale. However, some operations might need less (or more) precision than others to achieve the final desired precision. As such a smaller (or larger) scaling factor are used across the circuit between each level to optimize the total modulus size (make it as small as possible). As you pointed out, this the case in the bootstrapping, where for a final fixed precision, the SlotsToCoeffs doesn't require as much precision than the CoeffsToSlots, and thus uses smaller scaling factors.

This [paper](https://eprint.iacr.org/2020/1118) might give you some insights on the topic.
